skimage/restoration/_denoise_cy.pyx

```diff
@@ -176,22 +133,21 @@ def _denoise_tv_bregman(image, double weight, int max_iter, double eps,
         Py_ssize_t total = rows * cols * dims
 
     shape_ext = (rows2, cols2, dims)
-    u = np.zeros(shape_ext, dtype=np.double)
+    u = out.copy()
```

Comment:
By copying out, out doesn't become the output paramter anymore. And you are allocating a huge array without using it.

`u` should be removed in favour of `out` and other variables should be renamed accordingly.

Reply:
Thee reason this wasn't caught, is that the function "seemed to mutate out", but in fact, it was simply returning the correct value.